src/did/methods/ipid.js

```diff
@@ -0,0 +1,46 @@
+import createDidIpid from 'did-ipid';
+
+class Ipid {
+    static info = {
+        method: 'ipid',
+        description: 'The Interplanetary Identifiers DID method (IPID) supports DIDs on the public and private Interplanetary File System (IPFS) networks.',
+        homepageUrl: 'https://did-ipid.github.io/ipid-did-method/',
+        icons: [],
+    };
+
+    #didIpid;
+
+    constructor(didIpid) {
+        this.#didIpid = didIpid;
+    }
+
+    async resolve(did) {
+        return this.#didIpid.resolve(did);
+    }
+
+    async create(params, operations) {
+        const { privateKey } = { ...params };
+
+        return this.#didIpid.create(privateKey, operations);
+    }
+
+    async update(did, params, operations) {
+        const { privateKey } = { ...params };
+
+        return this.#didIpid.update(privateKey, operations);
+    }
+
+    async isPublicKeyValid(did, publicKeyId) {
+        const { publicKey = [] } = await this.resolve(did);
+
+        return publicKey.some((key) => key.id === publicKeyId);
+    }
+}
+
+const createIpid = async (ipfs) => {
```

Comment:
Perhaps we could make this factory synchronous and have the `resolve`, `create` and `update` create a `did-ipid` instance (once).

By doing this, the `Did` class dos not need to `await` for this factory.

Reply:
Done 👍